utils/interval.hh

```diff
@@ -781,6 +781,9 @@ public:
     explicit interval(const wrapping_interval<T>& r)
         : _interval(r)
     { }
+    explicit interval(const T& start, const T& end, bool start_inclusive = true, bool end_inclusive = true, bool singular = false)
+        : _interval(bound(start, start_inclusive), bound(end, end_inclusive), singular)
```

Comment:
singular=true is used in `interval_data(optional<bound> start, optional<bound> end, bool singular = false)` when the end bound is disengaged.
In this case, since both start and end must be provided I see no point in providing the `singular` parameter at all, but to hard code it as `false`.
(We can be smart and infer it from start==end, and both being inclusive, but we need not go there...)

Reply:
I guess that the inclusiveness parameters can be omitted as well.
If anyone wants to control inclusiveness, they can use the `optional<bound>` based ctor.